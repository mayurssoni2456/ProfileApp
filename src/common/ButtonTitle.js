import React from 'react';
import { TouchableOpacity, View, Text } from 'react-native';
import { ApplicationStyle } from '../Themes';

export default function ButtonTitle({ title, onPress, style, disabled }) {
  return (
    <View style={[ApplicationStyle.navButton, style]}>
      <TouchableOpacity
        disabled={!disabled}
        style={[{ justifyContent: 'center' }]}
        onPress={onPress}>
        <Text style={ApplicationStyle.navTextButton}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
}

/* validate = val => {
    // console.log(validateForm);
    if (validator.isMobilePhone(cell)) {
      // setCellValid(true);
      setValidateForm({ ...validateForm, cellValid: true });
    } else {
      // setCellValid(false);
      setValidateForm({ ...validateForm, cellValid: false });
    }

    setTimeout(() => {
      console.log(validateForm);
      if (validator.isEmail(email)) {
        // setEmailValid(true);
        setValidateForm({ ...validateForm, emailValid: true });
      } else {
        // setEmailValid(false);
        setValidateForm({ ...validateForm, emailValid: false });
      }
    }, 1000);
  }; */

/* const [validateForm, setValidateForm] = useState({
    cellValid: true,
    emailValid: true,
    cityValid: true,
    postalCodeValid: true,
  }); */
