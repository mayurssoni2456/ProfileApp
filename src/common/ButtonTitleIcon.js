import React from 'react';
import { TouchableOpacity, Image, View, Text } from 'react-native';
import { Icons } from '../helpers/imageMapper';
import { ApplicationStyle, Colors } from '../Themes';

export default function ButtonTitleIcon({ title, onPress, style }) {
  return (
    <View style={[style]}>
      <TouchableOpacity style={ApplicationStyle.navButton} onPress={onPress}>
        <Text style={ApplicationStyle.navTextButton}>{title}</Text>
        <Image source={Icons.rightArrow} style={ApplicationStyle.icon} />
      </TouchableOpacity>
    </View>
  );
}
