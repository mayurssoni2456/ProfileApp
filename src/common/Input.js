import React from 'react';
import { TextInput, View, Text } from 'react-native';
import { ApplicationStyle, Colors } from '../Themes';

const { fontSize, padding } = ApplicationStyle;

const classStyle = {
  title: {
    fontSize: fontSize.xMedium,
    paddingBottom: padding.sm,
    color: Colors.textGrey,
  },
  input: {
    borderBottomColor: Colors.borderGrey,
    borderBottomWidth: 1,
    fontSize: fontSize.xMedium,
    paddingVertical: padding.xSmall,
  },
};

export default function Input({
  title,
  value,
  onChangeText,
  style,
  isValid = true,
}) {
  return (
    <View style={[{ paddingVertical: padding.small }, style]}>
      <Text style={classStyle.title}>{title}</Text>
      <TextInput
        value={value}
        onChangeText={onChangeText}
        style={
          isValid
            ? classStyle.input
            : [classStyle.input, { borderBottomColor: 'red' }]
        }
      />
    </View>
  );
}
