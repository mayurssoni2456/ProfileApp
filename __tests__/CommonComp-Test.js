import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import ButtonTitle from '../src/common/ButtonTitle';
import ButtonTitleIcon from '../src/common/ButtonTitleIcon';
import Input from '../src/common/Input';

test('ButtonTitle Component Testing', () => {
  const tree = renderer.create(<ButtonTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('ButtonTitleIcon Component Testing', () => {
  const tree = renderer.create(<ButtonTitleIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Input Component Testing', () => {
  const tree = renderer.create(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});
