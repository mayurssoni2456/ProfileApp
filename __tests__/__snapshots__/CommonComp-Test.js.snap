// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTitle Component Testing 1`] = `
<View
  style={
    Array [
      Object {
        "borderColor": "rgba(188,188,188,1)",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "borderRadius": 2,
          "color": "rgba(83,83,83,1)",
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    />
  </View>
</View>
`;

exports[`ButtonTitleIcon Component Testing 1`] = `
<View
  style={
    Array [
      undefined,
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "borderColor": "rgba(188,188,188,1)",
        "borderRadius": 3,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "borderRadius": 2,
          "color": "rgba(83,83,83,1)",
          "flexDirection": "row",
          "fontSize": 18,
          "fontWeight": "600",
        }
      }
    />
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/rightArrow.png",
        }
      }
      style={
        Object {
          "height": 22,
          "width": 10,
        }
      }
    />
  </View>
</View>
`;

exports[`Input Component Testing 1`] = `
<View
  style={
    Array [
      Object {
        "paddingVertical": 10,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "color": "rgba(83,83,83,1)",
        "fontSize": 16,
        "paddingBottom": 10,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Object {
        "borderBottomColor": "rgba(188,188,188,1)",
        "borderBottomWidth": 1,
        "fontSize": 16,
        "paddingVertical": 5,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
